neo/rawio/spykingcircusrawio.py

```diff
@@ -0,0 +1,187 @@
+"""
+
+
+"""
+
+from .baserawio import (BaseRawIO, _signal_channel_dtype, _unit_channel_dtype,
+                        _event_channel_dtype)
+
+import numpy as np
+
+from pathlib import Path
+
+try:
+    import h5py
+    HAVE_HDF5= True
+except ImportError:
+    HAVE_HDF5 = False
+
+
+def _load_sample_rate(params_file):
+    sample_rate = None
+    with params_file.open('r') as f:
+        for r in f.readlines():
+            if 'sampling_rate' in r:
+                sample_rate = r.split('=')[-1]
+                if '#' in sample_rate:
+                    sample_rate = sample_rate[:sample_rate.find('#')]
+                sample_rate = float(sample_rate)
+    return sample_rate
+
+
+class SpykingCircusRawIO(BaseRawIO):
+    """
+
+    """
+    extensions = []
+    rawmode = 'one-folder'
+
+    def __init__(self, dirname=''):
+        BaseRawIO.__init__(self)
+        self.dirname = dirname
+
+    def _source_name(self):
+        return self.dirname
+
+    def _parse_header(self):
+        assert HAVE_HDF5, self.installation_mes
+        spykingcircus_folder = Path(self.dirname)
+        SpykingCircusRawIO.__init__(self, spykingcircus_folder)
+        listfiles = spykingcircus_folder.iterdir()
+        results = None
+        sample_rate = None
+
+        parent_folder = None
+        result_folder = None
+        for f in listfiles:
+            if f.is_dir():
+                if any([f_.suffix == '.hdf5' for f_ in f.iterdir()]):
+                    parent_folder = spykingcircus_folder
+                    result_folder = f
+
+        if parent_folder is None:
+            parent_folder = spykingcircus_folder.parent
+            for f in parent_folder.iterdir():
+                if f.is_dir():
+                    if any([f_.suffix == '.hdf5' for f_ in f.iterdir()]):
+                        result_folder = spykingcircus_folder
+
+        assert isinstance(parent_folder, Path) and isinstance(result_folder, Path), "Not a valid spyking circus folder"
+
+        # load files
+        for f in result_folder.iterdir():
+            if 'result.hdf5' in str(f):
+                results = f
+            if 'result-merged.hdf5' in str(f):
+                results = f
+                break
+
+        # load params
+        for f in parent_folder.iterdir():
+            if f.suffix == '.params':
+                sample_rate = _load_sample_rate(f)
+
+        if sample_rate is not None:
+            self._sampling_frequency = sample_rate
+
+        if results is None:
+            raise Exception(spykingcircus_folder, " is not a spyking circus folder")
+        f_results = h5py.File(results, 'r')
+
+        self._all_spikes = []
+        self.unit_labels = []
+        for temp in f_results['spiketimes'].keys():
+            self._all_spikes.append(np.array(f_results['spiketimes'][temp]).astype('int64'))
+            self.unit_labels.append(int(temp.split('_')[-1]))
```

Comment:
You should not load all then now but on demand

Reply:
Only unit_index now